docs/developer-guide/nodejs-api.md

```diff
@@ -17,22 +31,363 @@ While ESLint is designed to be run on the command line, it's possible to use ESL
+
+#### Return Value
+
+* (`Promise<Object>`)<br>
+  The promise that will be fulfilled with a configuration object.
+
+### ◆ eslint.isPathIgnored(filePath)
+
+```js
+const isPathIgnored = await eslint.isPathIgnored(filePath);
+```
+
+This method checks if a given file is ignored by your configurations.
+
+#### Parameters
+
+* `filePath` (`string`)<br>
+  The path to the file you want to check.
+
+#### Return Value
+
+* (`Promise<boolean>`)<br>
+  The promise that will be fulfilled with whether the file is ignored or not. If the file is ignored then `true`.
+
+### ◆ eslint.loadFormatter(nameOrPath)
+
+```js
+const formatter = await eslint.loadFormatter(nameOrPath);
+```
+
+This method loads a formatter.
+
+#### Parameters
+
+* `nameOrPath` (`string | undefined`)<br>
+  The path to the file you want to check. The following values are allowed:
+    * `undefined`. In this case, loads the `"stylish"` built-in formatter.
+    * One of [built-in formatter names][builtin-formatters].
+    * A third-party formatter name. For examples:
+        * `"foo"` will load `eslint-formatter-foo`.
+        * `"@foo"` will load `@foo/eslint-formatter`.
+        * `"@foo/bar"` will load `@foo/eslint-formatter-bar`.
+    * A path to the file that defines a formatter. The path must contain one or more path separators (`/`) in order to distinguish if it's a path or not.
+
+#### Return Value
+
+* (`Promise<Formatter>`)<br>
+  The promise that will be fulfilled with a [Formatter] object.
+
+### ◆ ESLint.version
+
+```js
+const version = ESLint.version;
+```
+
+The version string of ESLint. E.g. `"7.0.0"`.
+
+This is not instance property.
+
+### ◆ ESLint.outputFixes(results)
+
+```js
+await ESLint.outputFixes(results);
+```
+
+This method writes the modified code by the autofix feature into each file. If any modified files don't exist, this method does nothing.
+
+This is not instance method.
+
+#### Parameters
+
+* `results` (`LintResult[]`)<br>
+  The [LintResult] objects to write.
+
+#### Return Value
+
+* (`Promise<void>`)<br>
+  The promise that will be fulfilled after all files are written.
+
+### ◆ ESLint.getErrorResults(results)
+
+```js
+const filteredResults = ESLint.getErrorResults(results);
+```
+
+This method copies the given results and removes warnings. The returned value contains only errors.
+
+This is not instance method.
+
+#### Parameters
+
+* `results` (`LintResult[]`)<br>
+  The [LintResult] objects to filter.
+
+#### Return Value
+
+* (`LintResult[]`)<br>
+  The filtered [LintResult] objects.
+
+### ◆ LintResult type
+
+The `LintResult` value is the information of the linting result of each file. The [`eslint.lintFiles()`][eslint-lintfiles] and [`eslint.lintText()`][eslint-linttext] methods return it. It has following properties:
+
+* `filePath` (`string`)<br>
+  The absolute path to the file of this result. This may be the string `"<text>"` rather than a file path if the [`eslint.lintText()`][eslint-linttext] method returned it.
```

Comment:
Can we add an explanation of what `"<text>"` means (i.e. that it's a placeholder)? Or maybe link to some existing documentation if we already have some?

Reply:
I added a description.